packages/notion-utils/src/get-page-table-of-contents.ts

```diff
@@ -21,9 +21,14 @@ const indentLevels = {
 export const getPageTableOfContents = (
   recordMap: types.ExtendedRecordMap
 ): Array<TableOfContentsEntry> => {
-  const toc = (Object.keys(recordMap?.block) ?? [])
+  let recordMapBlock: any = []
+  if (recordMap && recordMap.block) {
+    recordMapBlock = recordMap.block
+  }
+  
+  const toc = (Object.keys(recordMapBlock) ?? [])
     .map((blockId: string) => {
-      const block = recordMap.block[blockId]?.value
+      const block = recordMapBlock[blockId]?.value
```

Comment:
Doesn't the coalescing operator already handle this? the recordmap array  will just be null if it is null.

Reply:
yeah that was my understanding as well, but maybe it doesn't compile correctly on production? Here's a live example of the issue. It seems to flash the content correctly then shows the error page.

https://www.notionpotions.space/
Error:
![image](https://user-images.githubusercontent.com/3444220/127039157-e163d49d-2c20-4af9-b320-c607eea9fe46.png)


here's how the compiled code looks like:
![image](https://user-images.githubusercontent.com/3444220/127038477-7b1133bf-bafb-40ed-bced-5247874aa0ea.png)

To me it looks like it checks to see whether `e` (or recordMap) is available and if so run `Object.keys(e.block)`. `e.block` can be undefined which will throw the error since Object.keys requires an array. Also looking at it again, it seems like if recordMap is undefined, it will throw an error as well.

Reply:
A better solution could be `(Object.keys(recordMap?.block ?? []) ?? [])`